src/reducers/scrollState.js

```diff
@@ -72,14 +71,18 @@ function scrollState(state = DEFAULT_STATE, action) {
 
       if (newProps.scrollTop && newProps.scrollTop !== oldProps.scrollTop) {
         state = scrollTo(state, newProps.scrollTop);
-        state = updateVisibleRows(state);
       }
 
       if (newProps.scrollToRow && newProps.scrollToRow !== oldProps.scrollToRow) {
         state = scrollToRow(state, newProps.scrollToRow);
-        state = updateVisibleRows(state);
       }
 
+      if (newProps.rowsCount !== state.rowsCount) {
+        state = updateRowCount(state, newProps);
+      }
+
+      state = updateVisibleRows(state);
```

Comment:
This is also required for the filter fixes, but I saw no obvious comparison to wrap it with based on props / state as the two checks above..